core/src/main/java/org/infinispan/remoting/inboundhandler/GlobalInboundInvocationHandler.java

```diff
@@ -153,7 +154,7 @@ private void runReplicableCommand(ReplicableCommand command, Reply reply, boolea
       try {
          invokeReplicableCommand(command, reply, preserveOrder);
       } catch (Throwable throwable) {
-         if (throwable.getCause() != null && throwable instanceof CompletionException) {
+         if (throwable.getCause() != null && (throwable instanceof CompletionException || throwable instanceof CacheConfigurationException)) {
```

Comment:
I would prefer changing `AbstractComponentRegistry.wireDependencies()` to stop wrapping `IllegalLifecycleStateException`.